### PR TITLE
Assassin Quest

### PR DESCRIPTION
**Pull request by xdeimos**
Adding the Assassin Quest script

---

requires https://github.com/themanaworld/tmwa-client-data/pull/302

### CODE REVIEW

File: world/map/npc/005-3/nysaa.txt [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+    if ((QL_ASSASSIN == 6) && (BaseLevel >= 77)) goto L_AssaPants;
+    if ((QL_ASSASSIN == 7) && (BaseLevel >= 77)) goto L_AssaPantsFinished;
+
+    if ((QL_ASSASSIN == 8) && (BaseLevel >= 78)) goto L_AssaShirt;
+    if ((QL_ASSASSIN == 9) && (BaseLevel >= 78)) goto L_AssaShirtFinished;
+    if ((QL_ASSASSIN == 10) && (BaseLevel >= 78)) goto L_Finished;
+
+    if ((BaseLevel >= 75) && ((getskilllv(SKILL_SPEED) == 9)) && (QL_ASSASSIN == 0))  goto L_Start;
+
+    mes "Nysaa";
```
Comment (mekolat) [inline world/map/npc/005-3/nysaa.txt]: here it's missing the `[ ]`

File: world/map/npc/005-3/nysaa.txt [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+        "Yes, I'm not busy.", L_Help,
+        "I'm sorry I cant.", L_No,
+        "What am I getting if I help you?", L_Interested;
+
+L_Help:
+    mes "[Nysaa]";
+    mes "\"We have had to make a lot of new equipment for the young members of the League.\"";
+    next;
+    mes "\"The tailors will need lots fo reptile skins, colored dyes, and the best silk in Kaizei.\"";
+    set QL_ASSASSIN, 1;
```
Comment (mekolat) [inline world/map/npc/005-3/nysaa.txt]: this line should be moved right at the beginning of L_Help

File: world/map/npc/005-3/nysaa.txt [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    mes "[Nysaa]";
+    mes "\"You see, I'm a member of the League, if you help us with the needed materials I'll make sure you get a nice reward.\"";
+    next;
+    mes "\"Okay so first of all the tailors will be crafting some gloves.\"";
+    next;
+    mes "\"Are you interested?\"";
+    menu
+        "Yes I'm.", L_Help,
+        "No", L_No,
+        "Who are you saying you are...?", L_Interested;
```
Comment (mekolat) [inline world/map/npc/005-3/nysaa.txt]: hm, you make `L_Interested` go to `L_Interested` ?
at this point you should just have yes and no

File: world/map/npc/005-3/nysaa.txt [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+
+L_AssaGloves:
+    if ( (countitem ("BlackCottonGloves") < 1)
+        ||(countitem ("SnakeSkin") < 50)
+        ||(countitem ("RedDye") < 15)
+        ||(countitem ("SilkCocoon") < 150) )
+           goto L_GlovesReq;
+    set QL_ASSASSIN, 2;
+
+    getinventorylist;
```
Comment (mekolat) [inline world/map/npc/005-3/nysaa.txt]: this line is not necessary here

File: world/map/npc/005-3/nysaa.txt [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+        "No", L_No,
+        "Who are you saying you are...?", L_Interested;
+
+L_AssaGloves:
+    if ( (countitem ("BlackCottonGloves") < 1)
+        ||(countitem ("SnakeSkin") < 50)
+        ||(countitem ("RedDye") < 15)
+        ||(countitem ("SilkCocoon") < 150) )
+           goto L_GlovesReq;
+    set QL_ASSASSIN, 2;
```
Comment (mekolat) [inline world/map/npc/005-3/nysaa.txt]: this line should be moved after the delitems

File: world/map/npc/005-3/nysaa.txt [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+
+    getinventorylist;
+    delitem "BlackCottonGloves", 1;
+    delitem "SnakeSkin", 50;
+    delitem "RedDye", 15;
+    delitem "SilkCocoon", 150;
+    mes "\"Whoa you have exactly every thing I asked you to bring\"";
```
Comment (mekolat) [inline world/map/npc/005-3/nysaa.txt]: here you should also say something like "come back later" or you could also directly `next;` and `goto L_AssaGlovesFinished;`

File: world/map/npc/005-3/nysaa.txt [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+    close;
+
+
+L_AssaGlovesFinished:
+    getinventorylist;
+    if ((checkweight("AssassinBoots", 1) == 0) || (@inventorylist_count == 100))
```
Comment (mekolat) [inline world/map/npc/005-3/nysaa.txt]: here you should check for the gloves instead of the boots

File: world/map/npc/005-3/nysaa.txt [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+        goto L_InventoryNoSpace;
+    mes "[Nysaa]";
```
Comment (mekolat) [inline world/map/npc/005-3/nysaa.txt]: remove that line

File: world/map/npc/005-3/nysaa.txt [hunk 9]
```diff
@@ -0,0 +1,263 @@
...
+    mes "\"The list says that we need 50 Snake Skins, 1 Black Dye, and 150 Silk Cocoons. And don't forget to bring me a pair of Black Boots.\"";
+    close;
+
+L_AssaBootsStart:
+    if ( (countitem("BlackCottonBoots") < 1)
+        || (countitem("BlackDye") < 1)
+        || (countitem("SnakeSkin") < 50)
+        || (countitem("SilkCocoon") < 150) )
+            goto L_BootsReq;
+    set QL_ASSASSIN, 5;
```
Comment (mekolat) [inline world/map/npc/005-3/nysaa.txt]: move this after the delitems

File: world/map/npc/005-3/nysaa.txt [hunk 10]
```diff
@@ -0,0 +1,263 @@
...
+        "No.", L_No;
+
+L_Okay:
+    if ( (countitem("BlackCottonTrousers") < 1)
+        || (countitem("RedDye") < 30)
+        || (countitem("SnakeSkin") < 100)
+        || (countitem("SilkCocoon") < 300) )
+            goto L_PantsReq;
+
+    set QL_ASSASSIN, 7;
```
Comment (mekolat) [inline world/map/npc/005-3/nysaa.txt]: after the delitems

**Comment by mekolat** [comment]
I'll check the rest later

**Comment by wushin** [comment]
https://forums.themanaworld.org/viewtopic.php?p=150646#p150646 Typo

**Pull request closed without merging**